api/v1alpha2/awsmachine_conversion.go

```diff
@@ -23,48 +23,50 @@ import (
 )
 
 // ConvertTo converts this AWSMachine to the Hub version (v1alpha3).
-func (src *AWSMachine) ConvertTo(dstRaw conversion.Hub) error { // nolint
+func (src *AWSMachine) ConvertTo(dstRaw conversion.Hub) error {
 	dst := dstRaw.(*infrav1alpha3.AWSMachine)
-	return Convert_v1alpha2_AWSMachine_To_v1alpha3_AWSMachine(src, dst, nil)
+	if err := Convert_v1alpha2_AWSMachine_To_v1alpha3_AWSMachine(src, dst, nil); err != nil {
+		return err
+	}
+
+	return nil
 }
 
 // ConvertFrom converts from the Hub version (v1alpha3) to this version.
-func (dst *AWSMachine) ConvertFrom(srcRaw conversion.Hub) error { // nolint
+func (dst *AWSMachine) ConvertFrom(srcRaw conversion.Hub) error {
 	src := srcRaw.(*infrav1alpha3.AWSMachine)
 	return Convert_v1alpha3_AWSMachine_To_v1alpha2_AWSMachine(src, dst, nil)
 }
 
 // ConvertTo converts this AWSMachineList to the Hub version (v1alpha3).
-func (src *AWSMachineList) ConvertTo(dstRaw conversion.Hub) error { // nolint
+func (src *AWSMachineList) ConvertTo(dstRaw conversion.Hub) error {
 	dst := dstRaw.(*infrav1alpha3.AWSMachineList)
 	return Convert_v1alpha2_AWSMachineList_To_v1alpha3_AWSMachineList(src, dst, nil)
 }
 
 // ConvertFrom converts from the Hub version (v1alpha3) to this version.
-func (dst *AWSMachineList) ConvertFrom(srcRaw conversion.Hub) error { // nolint
+func (dst *AWSMachineList) ConvertFrom(srcRaw conversion.Hub) error {
 	src := srcRaw.(*infrav1alpha3.AWSMachineList)
 	return Convert_v1alpha3_AWSMachineList_To_v1alpha2_AWSMachineList(src, dst, nil)
 }
 
 // Convert_v1alpha3_AWSMachineSpec_To_v1alpha2_AWSMachineSpec converts from the Hub version (v1alpha3) of the AWSMachineSpec to this version.
 // Requires manual conversion as infrav1alpha3.AWSMachineSpec.ImageLookupBaseOS does not exist in AWSMachineSpec.
-func Convert_v1alpha3_AWSMachineSpec_To_v1alpha2_AWSMachineSpec(in *infrav1alpha3.AWSMachineSpec, out *AWSMachineSpec, s apiconversion.Scope) error { // nolint
+func Convert_v1alpha3_AWSMachineSpec_To_v1alpha2_AWSMachineSpec(in *infrav1alpha3.AWSMachineSpec, out *AWSMachineSpec, s apiconversion.Scope) error {
 	if err := autoConvert_v1alpha3_AWSMachineSpec_To_v1alpha2_AWSMachineSpec(in, out, s); err != nil {
 		return err
 	}
 
 	// Manually convert FailureDomain to AvailabilityZone.
-	if in.FailureDomain != nil && out.AvailabilityZone == nil {
-		out.AvailabilityZone = in.FailureDomain
-	}
+	out.AvailabilityZone = in.FailureDomain
 
 	// Discards ImageLookupBaseOS
```

Comment:
Tangential, but it looks like we'll need to add backup/restore functionality similar to what was done in CAPI for this field (not sure about others).

Reply:
Yeah see comment above

Reply:
I saw another PR that was adding the restore logic, I'll create an issue to track handling this field.

Reply:
happy to follow up after both this and mine are merged.

Reply:
Created #1510 to track